x-pack/legacy/plugins/siem/public/containers/timeline/all/index.tsx

```diff
@@ -36,43 +36,43 @@ interface OwnProps extends AllTimelinesVariables {
   children?: (args: AllTimelinesArgs) => React.ReactNode;
 }
 
-const getAllTimeline = (variables: string, timelines: TimelineResult[]): OpenTimelineResult[] =>
-  timelines.map(timeline => ({
-    created: timeline.created,
-    description: timeline.description,
-    eventIdToNoteIds:
-      timeline.eventIdToNoteIds != null
-        ? timeline.eventIdToNoteIds.reduce((acc, note) => {
-            if (note.eventId != null) {
-              const notes = getOr([], note.eventId, acc);
-              return { ...acc, [note.eventId]: [...notes, note.noteId] };
-            }
-            return acc;
-          }, {})
-        : null,
-    favorite: timeline.favorite,
-    noteIds: timeline.noteIds,
-    notes:
-      timeline.notes != null
-        ? timeline.notes.map(note => ({ ...note, savedObjectId: note.noteId }))
-        : null,
-    pinnedEventIds:
-      timeline.pinnedEventIds != null
-        ? timeline.pinnedEventIds.reduce(
-            (acc, pinnedEventId) => ({ ...acc, [pinnedEventId]: true }),
-            {}
-          )
-        : null,
-    savedObjectId: timeline.savedObjectId,
-    title: timeline.title,
-    updated: timeline.updated,
-    updatedBy: timeline.updatedBy,
-  }));
+const getAllTimeline = memoizeOne(
```

Comment:
nit: Maybe a comment here to the distinction in use of `memoizeOne` vs `useMemo`? Is there any difference here since it's a function internal to the component?

>So, memoize-one (in common with most memoization libraries) remembers the result of a given function with a set of arguments, despite where the last execution occurs. React.memo, on the other hand, is for memoizing a single occurrence of a component when attempting to re-render, and it will not work outside of its instance.
Before we end this quick glance, I want to point out that there is a new useMemo hook, which works similarly to memoize-one, and is intended to memoize given functions in the context of a functional component. In fact, it can be used in conjunction with React.memo, to both memoize the component and any internal functions. We can take a deeper dive another day, but I hope that clears up a bit of confusion.

From: https://medium.com/better-programming/react-memo-vs-memoize-71f85eb4e1a#eb1f